apps/timetable/admin/ui/js/index.js

```diff
@@ -0,0 +1,411 @@
+
+            // Clear local storage
+            gh.api.utilAPI.localDataStorage().remove('admin.collapsed');
+
+            // Add the current collapsed module(s) back to the local storage
+            collapsedIds = _.compact([$('.gh-list-group-item-open').attr('data-id')]);
+            gh.api.utilAPI.localDataStorage().store('admin.collapsed', collapsedIds);
+        });
+    };
+
+    /**
+     * Set up the part picker in the subheader
+     *
+     * @param  {Event}     ev      Standard jQuery event
+     * @param  {Object}    data    Data object describing the selected tripos to fetch parts for
+     * @private
+     */
+    var setUpPartPicker = function(ev, data) {
+        var triposId = parseInt(data.selected, 10);
+
+        // Get the parts associated to the selected tripos
+        var parts = _.filter(triposData.parts, function(part) {
+            return parseInt(data.selected, 10) === part.ParentId;
+        });
+
+        // Render the results in the part picker
+        gh.api.utilAPI.renderTemplate($('#gh-subheader-part-template'), {
+            'data': parts
+        }, $('#gh-subheader-part'));
+
+        // Show the subheader part picker
+        $('#gh-subheader-part').show();
+
+        // Destroy the field if it's been initialised previously
+        $('#gh-subheader-part').chosen('destroy').off('change', setUpModules);
+
+        // Initialise the Chosen plugin on the part picker
+        $('#gh-subheader-part').chosen({
+            'no_results_text': 'No matches for',
+            'disable_search_threshold': 10
+        }).on('change', setUpModules);
+    };
+
+    /**
+     * Set up the Tripos picker in the subheader
+     *
+     * @private
+     */
+    var setUpTriposPicker = function() {
+        var triposPickerData = {
+            'courses': triposData.courses
+        };
+
+        _.each(triposPickerData.courses, function(course) {
+            course.subjects = _.filter(triposData.subjects, function(subject) {
+                return course.id === subject.ParentId;
+            });
+        });
+
+        // Massage the data so that courses are linked to their child subjects
+        // Render the results in the tripos picker
+        gh.api.utilAPI.renderTemplate($('#gh-subheader-picker-template'), {
+            'data': triposPickerData
+        }, $('#gh-subheader-tripos'));
+
+        // Show the subheader tripos picker
+        $('#gh-subheader-tripos').show();
+
+        // Destroy the field if it's been initialised previously
+        $('#gh-subheader-tripos').chosen('destroy').off('change', setUpModules);
+
+        // Initialise the Chosen plugin on the tripos picker
+        $('#gh-subheader-tripos').chosen({
+            'no_results_text': 'No matches for'
+        }).change(setUpPartPicker);
+    };
+
+    /**
+     * Show the tripos help info
+     *
+     * @private
+     */
+    var showTriposHelp = function() {
+        gh.api.utilAPI.renderTemplate($('#gh-tripos-help-template'), {
```

Comment:
The `tripos-help-template` contains information about the next step when you are logged in. The `help-template` contains a link for when logging in fails.